### DISCUSSION TITLE
Import plugin model into CustomScript

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#7795] [General]
Hey,

Need some advise.
I'm trying to import a Plugin's model into a CustomScript, but I'm failing to do so

I get this error
```
File "/opt/netbox/netbox/scripts/PopulateLiveData.py", line 8, in <module>
    from nb_sentia.models import IPAccounting
ImportError: cannot import name 'IPAccounting' from 'nb_sentia.models' (/opt/netbox/venv/lib/python3.7/site-packages/nb_sentia/models.py)
```

That works fine inside a nbshell, but not in a CustomScript.

What is the caveat ?

Pieter

**Comment by PieterL75** [comment]
Nevermind... After restarting netbox services it started working.. 
I guess my 'develop' loaded plugin is causing this

**Reply by DanSheps** [reply]
One quick note about plugins, if you have a error in one on runtime, it will not load and if you are developing it will not catch any edits for the autoreloader when this happens.

**Reply by jeremystretch** [reply]
I'd like to find out why this happens. FYI though if you enter the `nbshell` you should see the exception.

**Reply by PieterL75** [reply]
I run the ./manage.py runserver -v 3 0.0.0.0:8001 while I develop my plugins, and I have debug=true in the config...
That way I can see most of the exceptions..

I'm not able to replicate it anymore. 
Must have been a oneoff while I was adding the model and its functions (with a lot of bad programming and Pyhton N00B bugs)

Thx for the interest

**Reply by PieterL75** [reply]
I found the issue... I think..
My Plugin on git does not have the model IPAccounting yet. It uses CustomField to store the data.

Only the one on my develop folder has it.. 
It looks like the venv of the running netbox had the plugin installed from my git repo, as the folder refers to the venv site-package.
```/opt/netbox/venv/lib/python3.7/site-packages/nb_sentia/models.py```

the model,py with the IPAccounting model in, is in /home/pieter/nb_sentia/model.py

My 'restart' of netbox now includes uninstalling the plugin, install the develop one, and start the debug server... 
that should avoid this in the future

**Reply by DanSheps** [reply]
Yeah, like I said, when you are developing if there is a very critical error in the code and StatReloader just happens to reload it will not load the plugin.  Then it falls "out of scope" for StatReloader to care about since it isn't loaded so any additional edits to fix the error don't cause a reload and you need to manually restart the server.
